documentation/AuthoringSolutionsTutorial.md

```diff
@@ -0,0 +1,342 @@
+
+### Expressing Your Applications as a Document
+
+Integrating your solution, application, or other utility involves authoring a document using our structured solution schema formatted as JSON. In most cases you won't need to write any javascript code, as we have a number of plugins with syntax that are available from this schema. On occasion though, you may find something that can't be accomplished with the current schema and plugins, at which point you can work with our developers on creating the necessary extensions.
+
+The first step of this process involves thinking about what happens whenever someone uses your solution, and what happens if a different person needs to use it. For instance, your solution might be an application with an executable that must be launched, and stopped when the user is done.  And if a different user where to try it, they might need some of the settings changed which are different from the previous user.
+
+Also, you'll want to think about what features your solution has that are useful to users in general and also specifically in the domain of accessibility.
+
+With that in mind we'll dive in to the different sections of the solutions format, building up a complete example. The finished product is at the end of this document if you want to go down and reference it at any point. Also, a large number of examples for different platforms are available in the files in the directory `universal/testData/solutions`.
+
+### The Starting Template
+
+If you look at the files in `universal/testData/solutions` directory you'll see that a solutions file is a JSON object with each solution being another object keyed by the solutions unique id. For the JAWS example we'll use `com.freedomscientific.jaws` as the id. This id should be completely unique from any others, so it's useful to use the reverse domain naming system when possible.
+
+Our starting template looks as follows:
+
+```json
+    "com.freedomscientific.jaws": {
+        "name": "JAWS",
+        "contexts": {},
+        "settingsHandlers": [],
+        "lifecycleManager": {}
+    }
+```
+
+The solution are 4 sections.
+
+* name: This is a simple human readable name for the solution. We've filled it in with JAWS.
+* contexts: This specifices the context that your solution runs in, such as the operating system and any other dependencies it has.
+* settingsHandlers: The settings handlers sections describes where your solution stores it's settings, such as an XML file, or the windows registry, and how to change them. It can also map these settings to common terms that are used for accessibility needs and preferences.
```

Comment:
"windows registry" -> "Windows registry" (but "registry" is lower case in Microsoft's docs).